packages/flex-plugin-scripts/src/config/devServer/pluginServer.ts

```diff
@@ -129,10 +131,27 @@ export default (options: Configuration) => {
       return res.end('No X-Flex-JWE was provided');
     }
 
-    return _getRemotePlugins(jweToken, flexVersion)
+    const localInputPlugins = userInputPlugins.filter(p => !p.remote);
+    const remoteInputPlugins = userInputPlugins.filter(p => p.remote);
+
+    const hasRemotePlugin = includeAllRemote || userInputPlugins.some(p => p.remote);
+    const localPlugins = _getLocalPlugins(localInputPlugins);
+    const promise: Promise<Plugin[]> = hasRemotePlugin ? _getRemotePlugins(jweToken, flexVersion) : Promise.resolve([]);
+
+    // unnessary w/o --include-remote
+    return promise
+      .then(plugins => {
+        if (includeAllRemote) {
+          return plugins;
+        }
+
+        // Filter and only return the ones that are in remoteInputPlugins
+        return plugins.filter(r => remoteInputPlugins.some(i => i.name === r.name));
+      })
+      // rebase will eventually get both local and remote plugins
       .then(remotePlugins => {
         logger.trace('Got remote plugins', remotePlugins);
-        const plugins = _rebasePlugins(remotePlugins);
+        const plugins = _mergePlugins(localPlugins, remotePlugins);
 
         res.writeHead(200, responseHeaders);
         res.end(JSON.stringify(plugins));
```

Comment:
Just do in one line: `res.end(JSON.stringify(_mergePlugins(localPlugins, remotePlugins)))`